cmake/compiler/armclang/generic.cmake

```
@@ -6,6 +6,8 @@ find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}armclang PATHS ${TOOLCHAIN_HOME}/b
 
 set(triple arm-arm-none-eabi)
 
+find_package(armclang 6.17 REQUIRED)
```

It looks like we might want this to be `Armclang` or even `ArmClang` based on the existing naming convention.

```suggestion
find_package(Armclang 6.17 REQUIRED)
```

---

I had the name match the variable we use in ZEPHYR_TOOLCHAIN_VARIANT